addons/website_sale/models/sale_order.py

```diff
@@ -91,12 +91,7 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         else:
             domain += [('product_custom_attribute_value_ids', '=', False)]
 
-        lines = self.env['sale.order.line'].sudo().search(domain)
-
-        if line_id:
-            return lines
-
-        return self.env['sale.order.line']
+        return self.env['sale.order.line'].sudo().search(domain)
```

Comment:
`_cart_find_product_line` is old code patched several times that I could not fix due to how the models were set up in 12.0.

A complete revamp would be in order, this single change here does not seem enough to make it work correctly again.

I would only consider to change the method if it was completely fixed, with all cases unit tested, including:
- `product_id`
- `line_id` (there is actually a big distinction depending on if this is passed or not, including on the overrides)
- pass "no variant" values when applicable
- pass "custom" values when applicable
- override with `event_ticket_id` in context
- override with `linked_line_id` and/or `optional_product_ids`

I'm not even sure if we know what we should do in some of those cases, especially when several of those arguments are passed at the same time, so that has to be clarified functionally first too.

Reply:
Looked into it and a separate task should be created for this one It looks like there is indeed a little more to it than just what I did...
I will remove the commit and create another branch/PR later